synapse/handlers/sync.py

```diff
@@ -533,11 +592,23 @@ def compute_state_delta(self, room_id, batch, sync_config, since_token, now_toke
                 state_ids = _calculate_state(
                     timeline_contains=timeline_state,
                     timeline_start=state_at_timeline_start,
+                    timeline_start_members=member_state_ids,
                     previous=state_at_previous_sync,
                     current=current_state_ids,
                 )
             else:
                 state_ids = {}
+                if lazy_load_members:
+                    # TODO: filter out redundant members based on their mxids (not their
+                    # event_ids) at this point. We know we can do it based on mxid as this
+                    # is an non-gappy incremental sync.
+
+                    # strip off the (None, None) and filter to just room members
+                    types = types[:-1]
```

Comment:
rly? we've only just added the damn thing.

Reply:
fixed by `filtered_types`